### PR TITLE
Refactoring

### LINKED ISSUE #271: Big Refactoring
* [x] Refactor utilities
* [x] Refactor `phy.io`: `phy.io.mock`, `phy.io.kwik.model`, `phy.io.kwik.mock`
* [x] Move cluster store to `phy.io`
* [x] Move `features_masks` and `waveforms` store items to `phy.io.kwik`
* [x] `phy.plot.view_models` with `kwik` and `_base`

cc @nippoo

### PR DESCRIPTION
**Pull request by rossant**
Closes #271

### CODE REVIEW

File: phy/cluster/manual/clustering.py [hunk 1]
```diff
@@ -8,12 +8,13 @@
...
-                     _spikes_per_cluster,
-                     UpdateInfo,
-                     )
+from ...utils import (_as_array,
+                      _is_array_like,
+                      _unique,
+                      _spikes_in_clusters,
+                      _spikes_per_cluster,
+                      )
+from ._utils import UpdateInfo
```
Comment (nippoo) [inline phy/cluster/manual/clustering.py]: I know we've been through this before, but this is why this kind of relative import is really confusing!

Comment (rossant) [inline phy/cluster/manual/clustering.py]: boaf.. it wasn't as painful as i feared to refactor them

File: phy/plot/view_models/tests/test_kwik.py [hunk 2]
```diff
@@ -75,17 +75,17 @@ def _test_view_model(view_model_class, stop=True, do_cluster=False, **kwargs):
...
-        # Split some spikes and update the view.
-        debug("Splitting.")
-        spikes = spikes[::2]
-        up = clustering.assign(spikes, np.random.randint(low=0, high=5,
-                                                         size=len(spikes)))
-        vm.on_select(up.added)
-        show_test_run(vm.view, _N_FRAMES)
+        # up = clustering.merge(clusters)
+        # vm.on_select(up.added)
+        # show_test_run(vm.view, _N_FRAMES)
```
Comment (nippoo) [inline phy/plot/view_models/tests/test_kwik.py]: euhhhhhhhhh?

Comment (rossant) [inline phy/plot/view_models/tests/test_kwik.py]: the next pr will refactor all this stuff

**Pull request merged**